jetty-server/src/main/java/org/eclipse/jetty/server/handler/ResourceHandler.java

```diff
@@ -157,27 +163,28 @@ public Resource getResource(String path)
                 path = URIUtil.canonicalPath(path);
                 r = _baseResource.addPath(path);
 
-                if (r != null && r.isAlias() && (_context == null || !_context.checkAlias(path, r)))
+                if (r.isAlias() && (_context == null || !_context.checkAlias(path, r)))
                 {
                     if (LOG.isDebugEnabled())
-                        LOG.debug("resource={} alias={}", r, r.getAlias());
-                    return null;
+                        LOG.debug("Rejected alias resource={} alias={}", r, r.getAlias());
+                    throw new IOException("Rejected (see debug logs)");
                 }
             }
             else if (_context != null)
+            {
                 r = _context.getResource(path);
+                if (r != null)
+                    return r;
+            }
 
             if ((r == null || !r.exists()) && path.endsWith("/jetty-dir.css"))
                 r = getStylesheet();
 
-            return r;
-        }
-        catch (Exception e)
-        {
-            LOG.debug("Unable to get Resource for {}", path, e);
+            if (r != null)
+                return r;
         }
 
-        return null;
+        throw new IOException("Unable to find Resource for " + path);
```

Comment:
I don't think the message is right.
This method should still return a resource EVEN IF not found, as the request might be a PUT about to create the resource.
This method should only throw if something has gone horribly wrong.
Perhaps this should be an IllegalArguementException ?

Reply:
It cannot return a resource if the Resource itself cannot be created.

Some examples:

``` java
String resultA = getResource("foo#file"); // wont work on URLResource, should probably have been ecnoded.
String resultB = getResource("bar:zed"); // wont work on windows for PathResource
String resultC = getResource("WEB-INF/lib/foo.jar!/deep/resource"); // wont work on URLResource (but will work on PathResource)
```

If you insist on returning a Resource, perhaps we should instead of throwing `IOException` return `new BadResource()` instead?

Reply:
You've miss understood.     I'm not saying return a resource if it is bad.  
I'm saying that "Unable to find Resource" is the wrong message for a bad resource.
I think `IllegalArgumentException` is a possibility..... hmm except that those examples you give are legal, just not for the base resource..... so IOException is probably correct.... just that the message is wrong.

Reply:
still not right.  See new review.